pymatgen/io/lobster/lobsterenv.py

```diff
@@ -1154,15 +1166,22 @@ def _get_limit_from_extremum(
                 if (val1 < 0.0 < val2) or (val2 < 0.0 < val1):
                     list_icohps.append(value.summed_icohp)
 
-            extremum_based = min(list_icohps) * percentage
+            if not self.are_coops and not self.are_cobis:
+                extremum_based = min(list_icohps) * percentage
+            else:
+                extremum_based = max(list_icohps) * percentage
 
         elif additional_condition == 2:
             # NO_ELEMENT_TO_SAME_ELEMENT_BONDS
             list_icohps = []
             for value in icohpcollection._icohplist.values():
                 if value._atom1.rstrip("0123456789") != value._atom2.rstrip("0123456789"):
                     list_icohps.append(value.summed_icohp)
-            extremum_based = min(list_icohps) * percentage
+
+            if not self.are_coops and not self.are_cobis:
+                extremum_based = min(list_icohps) * percentage
+            else:
+                extremum_based = max(list_icohps) * percentage
```

Comment:
This code is repeated 5 times. Maybe turn into a function. Could definitely be made more concise:

```suggestion
            which_extr = min if not self.are_coops and not self.are_cobis else max
            extremum_based = which_extr(list_icohps) * percentage
```

Reply:
> This code is repeated 5 times. Maybe turn into a function. Could definitely be made more concise:

Thanks for this suggestion . I will implement this.